File: ControlDeHora.cpp
#include <iostream>
using namespace std;

int main() {

```
int hora, minutos, segundos;
char opcion;

do {
    cout << "Ingrese la hora: ";
    cin >> hora;

    cout << "Ingrese los minutos: ";
    cin >> minutos;

    cout << "Ingrese los segundos: ";
    cin >> segundos;

    cout << "Hora registrada: "
         << hora << ":" << minutos << ":" << segundos << endl;

    cout << "¿Desea cambiar la hora? (S/N): ";
    cin >> opcion;

} while (opcion == 'S' || opcion == 's');

cout << "Programa finalizado." << endl;

return 0;
```

}

File: ControlDeHora.java
import java.util.Scanner;

public class ControlDeHora {
public static void main(String[] args) {

    Scanner sc = new Scanner(System.in);

    int hora, minutos, segundos;
    char opcion;

    do {
        System.out.print("Ingrese la hora: ");
        hora = sc.nextInt();

        System.out.print("Ingrese los minutos: ");
        minutos = sc.nextInt();

        System.out.print("Ingrese los segundos: ");
        segundos = sc.nextInt();

        System.out.println("Hora registrada: " 
                + hora + ":" + minutos + ":" + segundos);

        System.out.print("¿Desea cambiar la hora? (S/N): ");
        opcion = sc.next().charAt(0);

    } while (opcion == 'S' || opcion == 's');

    System.out.println("Programa finalizado.");

    sc.close();
}

}
